lib/Model/Importer/Binary/MipsELFImporter.h

```diff
@@ -0,0 +1,103 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+using namespace llvm;
+
+template <typename T, bool HasAddend>
+class MipsELFImporter {
+  llvm::Optional<uint64_t> SymbolsCount;
+  llvm::Optional<uint64_t> MIPSFirstGotSymbol;
+  llvm::Optional<uint64_t> MIPSLocalGotEntries;
+
+ public:
+  bool importMipsDynSymbol(TupleTree<model::Binary> &Model, uint64_t Tag) {
+    assert((Model->Architecture == model::Architecture::mips or
+            Model->Architecture == model::Architecture::mipsel) and
+           "The interface must be used for MIPS binaries only");
```

Comment:
Please use `revng_assert`.
Also, make sure you're running `orc shell ./scripts/revng-check-conventions --force-format`, it will warn you about these things.